mirror images in bastion

sgaoshang opened the pull request.

sgaoshang commented:
/test all

sgaoshang commented:
/test all

jinyunma commented:
vsphere-upi-disconnected provision chain also need to be updated.
https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/installer/rehearse/vsphere/upi/disconnected/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-provision-chain.yaml

sgaoshang commented:
> vsphere-upi-disconnected provision chain also need to be updated. https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/installer/rehearse/vsphere/upi/disconnected/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-provision-chain.yaml

@jinyunma I see several workflows that may also need or not to change mirror steps, maybe we can change them when necessary, what do you think?

```
$ grep -R "mirror-images-by-oc-adm" * -B 1
ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml-  - chain: vsphere-provision-bastionhost
ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml:  - ref: mirror-images-by-oc-adm-in-bastion
--
ipi/disconnected/techpreview/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-techpreview-provision-chain.yaml-  - chain: vsphere-provision-bastionhost
ipi/disconnected/techpreview/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-techpreview-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
--
upi/disconnected/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-provision-chain.yaml-  - chain: vsphere-provision-bastionhost
upi/disconnected/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
--
upi/disconnected/sdn/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-sdn-provision-chain.yaml-  - chain: vsphere-provision-bastionhost
upi/disconnected/sdn/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-sdn-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
--
upi/disconnecting-network/provision/cucushift-installer-rehearse-vsphere-upi-disconnecting-network-provision-chain.yaml-  - ref: ipi-install-times-collection
upi/disconnecting-network/provision/cucushift-installer-rehearse-vsphere-upi-disconnecting-network-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
```

jianlinliu commented on ci-operator/step-registry/mirror-images/by-oc-adm-in-bastion/mirror-images-by-oc-adm-in-bastion-commands.sh:
```diff
@@ -0,0 +1,73 @@
+
+# mirror images in bastion host, which will increase mirror upload speed
+ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
+oc adm release -a /tmp/new_pull_secret mirror --insecure=true \
```
Because the `oc adm release` command is running on remote host, it is better to be quoted by "".

sgaoshang replied:
Fixed

jianlinliu commented on ci-operator/step-registry/mirror-images/by-oc-adm-in-bastion/mirror-images-by-oc-adm-in-bastion-commands.sh:
```diff
@@ -0,0 +1,73 @@
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+mirror_output="${SHARED_DIR}/mirror_output"
+new_pull_secret="${SHARED_DIR}/new_pull_secret"
```
Suggest to introduce a new var to save `new_pull_secret`, so that we can avoid hardcode on line 63.

sgaoshang replied:
Updated

jianlinliu commented:
> @jinyunma I see several workflows that may also need or not to change mirror steps, maybe we can change them when necessary, what do you think?

If we update one for them, maybe we can update others together:
```
[jianlinliu@jialiu-laptop rehearse]$ grep -r "mirror-images-by-oc-adm" |grep vsphere
vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
vsphere/ipi/disconnected/techpreview/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-techpreview-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
vsphere/upi/disconnected/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
vsphere/upi/disconnected/sdn/provision/cucushift-installer-rehearse-vsphere-upi-disconnected-sdn-provision-chain.yaml:  - ref: mirror-images-by-oc-adm
vsphere/upi/disconnecting-network/provision/cucushift-installer-rehearse-vsphere-upi-disconnecting-network-provision-chain.yaml:  - ref: mirror-images-by-oc-adm

```

sgaoshang commented on ci-operator/step-registry/mirror-images/by-oc-adm-in-bastion/mirror-images-by-oc-adm-in-bastion-commands.sh:
```diff
@@ -0,0 +1,75 @@
+mirror_output="${SHARED_DIR}/mirror_output"
+pull_secret_filename="new_pull_secret"
+new_pull_secret="${SHARED_DIR}/${pull_secret_filename}"
+remote_pull_secret="/tmp/${pull_secret_filename}"
```
gitleaks:allow

jianlinliu commented:
ci/prow/step-registry-metadata — Job failed. Others looks good to me, and leave this to @jinyunma for final approve.

jinyunma commented:
For workflow `cucushift-installer-rehearse-vsphere-upi-disconnecting-network`, due to issue of mirroring image to ibmc vsphere, I added dedicated cluster_profile `vsphere-connected` to limit job only running on VMC. 
Since issue is fixed in this PR, I think we can move cluster profile from `vsphere-connected` to `vsphere` when running job with workflow `cucushift-installer-rehearse-vsphere-upi-disconnecting-network`.

Could you help to remove relevant job definition(cluster profile is `vsphere-connected`) in verification-test job together when fixing step-registry-metadata failure?
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.11.yaml#L75-L79
https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/verification-tests/openshift-verification-tests-master-periodics.yaml#L1129-L1203

And verify it via updating workflow to `cucushift-installer-rehearse-vsphere-upi-disconnecting-network` in https://github.com/openshift/release/blob/master/ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.11.yaml#L74

jianlinliu commented:
btw, do not forget to squash multiple commits into one.

sgaoshang commented:
> For workflow `cucushift-installer-rehearse-vsphere-upi-disconnecting-network`, due to issue of mirroring image to ibmc vsphere, I added dedicated cluster_profile `vsphere-connected` to limit job only running on VMC. Since issue is fixed in this PR, I think we can move cluster profile from `vsphere-connected` to `vsphere` when running job with workflow `cucushift-installer-rehearse-vsphere-upi-disconnecting-network`.
> 
> Could you help to remove relevant job definition(cluster profile is `vsphere-connected`) in verification-test job together when fixing step-registry-metadata failure? https://github.com/openshift/release/blob/master/ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.11.yaml#L75-L79 https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/verification-tests/openshift-verification-tests-master-periodics.yaml#L1129-L1203
> 
> And verify it via updating workflow to `cucushift-installer-rehearse-vsphere-upi-disconnecting-network` in https://github.com/openshift/release/blob/master/ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.11.yaml#L74

Fixed, pls check again, thanks for the info. Plus should I also delete profile vsphere-connected in ci-tools?

jinyunma commented:
> Fixed, pls check again, thanks for the info. Plus should I also delete profile vsphere-connected in ci-tools?
It's better to let job run some time to see if any issue detected. If all works well, I guess we can delete cluster profile `vsphere-connected`later.

I saw that job `installer-rehearse-4.11-installer-rehearse-vsphere` with workflow `vsphere-upi-disconnecting-network` passed.  This is expected job to verify mirror image into ibmc vsphere.
But step `set-sample-operator-disconnected` seems also need to mirror image to dynamic bastion host, and mirror command was running in pod on VMC CI build cluster, it took 1h19m30s, not sure any potential issue although it is passed this time?

sgaoshang commented:
@jinyunma Sure, I won't delete profile vsphere-connected now.
step `set-sample-operator-disconnected` is use by lots of other platforms, I would suggest fix it in another PR.

jinyunma commented:
/lgtm

jianlinliu commented:
/lgtm
